### ISSUE TITLE
Wasm-bindgen does not pass String from Rust to JS

### ISSUE THREAD

**Issue by wdanilo** [rustwasm/wasm-bindgen#3224]
### Describe the Bug

Hi, I've got such a code:

```rust
#[wasm_bindgen]
pub fn test() -> String {
    "hello".to_string()
}
```

When compiling it with `wasm-pack --target=web` and calling it from JS, I've got `undefined` as a result. It seems that I'm not the only one facing this issue and it seems that compiling everything with `--target=nodejs` makes it working: https://stackoverflow.com/questions/70079979/how-do-i-return-a-string-from-a-rust-function-in-nodejs-webassembly-using-wasm-b

I'm not sure whether this issue is connected more with wasm-pack or wasm-bindgen, but as wam-pack is not very actively maintained, I prefer to ask here first for any help / guidance. Thank you :)

**Comment by Liamolucko** [comment]
It's hard to say what exactly the issue is without seeing how you're instantiating the wasm module on the JS side, as well as where you're running the JS, etc.

If you're doing the same thing as the Stack Overflow question you linked, using `WebAssembly.instantiate` on the raw wasm file, the problem is that you should be importing the JS wrapper around it instead. The API exported by the raw wasm isn't meant to be used directly, which is why it returns undefined instead of the string you expect.

**Comment by wdanilo** [comment]
You are right, the Stack Overflow code is wrong and I didn't notice it. I am using the JS wrapper. The setup here is pretty complex. Let me clean it a little bit and I link it with a proper description in a few days. Thank you for the help.

**Comment by jre0** [comment]
@wdanilo if you are still looking for a solution, please check out this gist:
[https://gist.github.com/jre0/6913e4376d59806faaff349ba009fda2](https://gist.github.com/jre0/6913e4376d59806faaff349ba009fda2)
I found that there are old examples that show to calling the function on the object returned by the promise resolve. This always returned undefined for me. But if you import your function directly from generated js, it works. See this documention:
[https://rustwasm.github.io/docs/wasm-bindgen/examples/without-a-bundler.html](https://rustwasm.github.io/docs/wasm-bindgen/examples/without-a-bundler.html)
Hope this helps.
